types/types.go

```diff
@@ -12,19 +12,56 @@ type Configuration struct {
 	Servers []Server `json:"servers"`
 }
 
-// RaftServer describes a single raft server
-type RaftServer struct {
+// LogData is an instance of a single log
+type LogData struct {
+	Term    int    `json:"term"`
+	Command string `json:"command"`
+}
+
+// State has all the data on the
+// servers state.
+type State struct {
 	// can be follower, leader or candidate
 	// all servers start as a follower, if they
 	// dont hear from a leader, they can become
 	// candidates. Leaders are elected from the
 	// leader election process.
-	State string `json:"state"`
-	IP    string `json:"IP"`
-	Port  string `json:"Port"`
+	Name        string
+	CurrentTerm int
+	// VotedFor maintains the ID of the voted
+	// server; -1 if its leader, -2 at init
```

Comment:
1. Are the IDs zero-indexed or one-indexed?
2. What's the advantage of using -1 instead of self ID?

Reply:
Spec says use null. So I followed it.